src/rawposix/tests/fs_test.rs

```diff
@@ -0,0 +1,187 @@
+use threei::cage::*;
+use threei::fdtables;
+use threei::rawposix::vmmap::*;
+use threei::threei::{threei::*, threeiconstant};
+
+use std::thread;
+use std::time::{Duration, Instant};
+use tracing::{info, instrument};
+use tracing_subscriber::{fmt, layer::SubscriberExt, util::SubscriberInitExt};
+
+const FDKIND_KERNEL: u32 = 0;
+
+/// Helper functions:
+/// Create a test cage for testing purpose
+fn simple_init_cage(cageid: u64) {
+    println!("simple_init_cage called with cageid: {}", cageid);
+    // fdtables::register_close_handlers(FDKIND_KERNEL, fdtables::NULL_FUNC, kernel_close);
+    let cage = Cage {
+        cageid: cageid,
+        cwd: RwLock::new(Arc::new(PathBuf::from("/"))),
+        parent: 1,
+        gid: AtomicI32::new(-1),
+        uid: AtomicI32::new(-1),
+        egid: AtomicI32::new(-1),
+        euid: AtomicI32::new(-1),
+        main_threadid: AtomicU64::new(0),
+        zombies: RwLock::new(vec![]),
+        child_num: AtomicU64::new(0),
+        vmmap: RwLock::new(Vmmap::new()),
+    };
+    add_cage(cage);
+    fdtables::init_empty_cage(cageid);
+    println!("ADDED cage {:?} to fdtable", cageid);
+    // Set the first 3 fd to STDIN / STDOUT / STDERR
+    // STDIN
+    // let dev_null = CString::new("/home/lind/lind_project/src/safeposix-rust/tmp/dev/null").unwrap();
+    fdtables::get_specific_virtual_fd(cageid, 0, FDKIND_KERNEL, 0, false, 0).unwrap();
+    // STDOUT
+    fdtables::get_specific_virtual_fd(cageid, 1, FDKIND_KERNEL, 1, false, 0).unwrap();
+    // STDERR
+    fdtables::get_specific_virtual_fd(cageid, 2, FDKIND_KERNEL, 2, false, 0).unwrap();
+}
+
+/// Test read syscall behavior
+#[test]
+fn test_read() {
+    let cageid = 52;
+    simple_init_cage(cageid);
+
+    let mut pipe_fds = [-1; 2];
+    pipe(cageid, pipe_fds.as_mut_ptr());
```

Comment:
pipe isnt in this PR right?

I'm wondering if it makes sense for the tests to not import * and have these be like fs_calls::pipe to make it clear that we're using the rawposix version?  (not sure if thats the correct path)